test/parallel/test-tls-socket-failed-handshake-emits-error.js

```diff
@@ -0,0 +1,38 @@
+'use strict';
+const common = require('../common');
+
+if (!common.hasCrypto) {
+  common.skip('missing crypto');
+  return;
+}
+const tls = require('tls');
+const net = require('net');
+const assert = require('assert');
+
+const bonkers = Buffer.alloc(1024, 42);
+
+const server = net.createServer(function(c) {
+  setTimeout(function() {
+    const s = new tls.TLSSocket(c, {
```

Comment:
this version of the test exists because the error handling code paths under test differ in behaviour depending on how the server is created, or just to be very thorough?